lib/authentication.js

```diff
@@ -68,30 +75,16 @@ function Authentication(socketAddress)
         // Validate function parameters
         checkCredentials(credentials);
 
-        return new Promise((resolve, reject) =>
+        let options = JSON.parse(JSON.stringify(_defaultOptions));
+        options.body = 'command=signIn&user=' + credentials.username + '&password=' + credentials.password;
+        return requestPromise.post(options, (error, response, body) =>
           {
-            let request = new XMLHttpRequest();
-
-            request.onreadystatechange = () =>
-              {
-                if (request.readyState === 4) // 4 == DONE
-                {
-                  if (request.status === 200) // 200 == OK
-                  {
-                    _sessionToken = JSON.parse(request.responseText)['data']['clientToken'];
-                    resolve();
-                  }
-                  else
-                  {
-                    // TODO: Parse error message instead of dumping a JSON error.
-                    reject(request.responseText);
-                  }
-                }
-              }
-
-            request.open('POST', _url);
-            request.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
-            request.send('command=signIn&user=' + credentials.username + '&password=' + credentials.password);
+            if (  !error
+               && (response.statusCode == 200)
+               && (body.data.clientToken !== 'undefined'))
```

Comment:
And also, check if `data` itself exists, otherwise the dereference of `clientToken` will lead to a JavaScript error.